[Mappings editor] Improvements to boolean, binary and ip copy

alisonelizabeth opened the pull request:
This PR updates the text for the `null_value` parameter for `boolean` and `ip` data types. The rest of the parameters have been updated via https://github.com/elastic/kibana/pull/53319.

It also adds description text for `boolean`, `ip` and `binary` data types.

Other related changes:
- I noticed `ip_range` wasn't in the list of range types, so I added it.

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/)
* Commit: cb3bf1db8af5411242fed81a7f1420128954a4fb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/","number":"16237","commit":"cb3bf1db8af5411242fed81a7f1420128954a4fb"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/)
* Commit: 2b652d2245cd36611fee6a1a0c828155c4753ea6

### History
* :green_heart: [Build #16237](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/) succeeded cb3bf1db8af5411242fed81a7f1420128954a4fb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/","number":"16237","commit":"cb3bf1db8af5411242fed81a7f1420128954a4fb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/","number":"16971","commit":"2b652d2245cd36611fee6a1a0c828155c4753ea6"}]}
PIPELINE-->

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/boolean_type.tsx:
```diff
@@ -70,7 +70,8 @@ export const BooleanType = ({ field }: Props) => {
             {
-              defaultMessage: 'Whether to substitute values for any explicit null values.',
+              defaultMessage:
+                'Replace explicit null values with the selected boolean value so that it can be indexed and searched.',
```
I think "the selected boolean value" lacks context when the toggle is disabled, since the select element is hidden. How about changing it to "a specific boolean value"?

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/constants/parameters_definition.tsx:
```diff
@@ -252,6 +252,16 @@ export const PARAMETERS_DEFINITION = {
+      defaultValue: '',
+      type: FIELD_TYPES.TEXT,
+      label: nullValueLabel,
+      helpText: i18n.translate('xpack.idxMgmt.mappingsEditor.parameters.nullValueHelpText', {
```
Nit: I know we don't have this key in use elsewhere, but maybe it should be `nullValueIpHelpText` to be more specific?

alisonelizabeth replied:
👍 good catch

alisonelizabeth commented:
@cjcenizal thanks for the review! Mind taking another look? I added descriptions for `boolean`, `ip` and `binary` (this one I had initially reviewed, but didn't require any other copy changes). I also noticed the `ip_range` type was missing, so I added it.

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/constants/data_types_definition.tsx:
```diff
@@ -249,6 +300,12 @@ export const TYPE_DEFINITION: { [key in DataType]: DataTypeDefinition } = {
+    label: i18n.translate('xpack.idxMgmt.mappingsEditor.dataType.ipRangeDescription', {
+      defaultMessage: 'IP range',
+    }),
+    value: 'ip_range',
```
Maybe this deserves a description too?

alisonelizabeth replied:
I think this can probably be added when we address any copy changes for the `range` data type.

cjcenizal replied:
That makes sense!

cjcenizal commented on x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/constants/data_types_definition.tsx:
```diff
@@ -158,6 +190,18 @@ export const TYPE_DEFINITION: { [key in DataType]: DataTypeDefinition } = {
+      <p>
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.dataType.booleanLongDescription"
+          defaultMessage="Boolean fields accept JSON {true} and {false} values, as well as strings which are interpreted true or false."
```
Nit: "interpreted _as_ true or false"

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17017/)
* Commit: a047b92377a9265d57a2f15a194c34c0ff4e6360

### History
* :broken_heart: [Build #16971](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/) failed 2b652d2245cd36611fee6a1a0c828155c4753ea6
* :green_heart: [Build #16237](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/) succeeded cb3bf1db8af5411242fed81a7f1420128954a4fb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/","number":"16237","commit":"cb3bf1db8af5411242fed81a7f1420128954a4fb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/","number":"16971","commit":"2b652d2245cd36611fee6a1a0c828155c4753ea6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17017/","number":"17017","commit":"a047b92377a9265d57a2f15a194c34c0ff4e6360"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17026/)
* Commit: 0940451e950f839282b294becde4277b73d99146

### History
* :broken_heart: [Build #17017](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17017/) failed a047b92377a9265d57a2f15a194c34c0ff4e6360
* :broken_heart: [Build #16971](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/) failed 2b652d2245cd36611fee6a1a0c828155c4753ea6
* :green_heart: [Build #16237](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/) succeeded cb3bf1db8af5411242fed81a7f1420128954a4fb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/","number":"16237","commit":"cb3bf1db8af5411242fed81a7f1420128954a4fb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/","number":"16971","commit":"2b652d2245cd36611fee6a1a0c828155c4753ea6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17017/","number":"17017","commit":"a047b92377a9265d57a2f15a194c34c0ff4e6360"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17026/","number":"17026","commit":"0940451e950f839282b294becde4277b73d99146"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17038/)
* Commit: b0e88dbfdf1179a2b6db20f7bfe1a0e7f8210b12

### History
* :broken_heart: [Build #17026](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17026/) failed 0940451e950f839282b294becde4277b73d99146
* :broken_heart: [Build #17017](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17017/) failed a047b92377a9265d57a2f15a194c34c0ff4e6360
* :broken_heart: [Build #16971](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/) failed 2b652d2245cd36611fee6a1a0c828155c4753ea6
* :green_heart: [Build #16237](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/) succeeded cb3bf1db8af5411242fed81a7f1420128954a4fb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16237/","number":"16237","commit":"cb3bf1db8af5411242fed81a7f1420128954a4fb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16971/","number":"16971","commit":"2b652d2245cd36611fee6a1a0c828155c4753ea6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17017/","number":"17017","commit":"a047b92377a9265d57a2f15a194c34c0ff4e6360"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17026/","number":"17026","commit":"0940451e950f839282b294becde4277b73d99146"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/17038/","number":"17038","commit":"b0e88dbfdf1179a2b6db20f7bfe1a0e7f8210b12"}]}
PIPELINE-->
